src/devices/srsmith_pool_srs_2c_tx.c

```diff
@@ -0,0 +1,175 @@
+/** @file
+    SRSmith Pool Light Remote Control, Model #SRS-2C-TX
+
+    Copyright (C) 2022 gcohen55
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "decoder.h"
+#include "util.h"
+//#include <inttypes.h>
+//#include <string.h>
+
+#define BUTTON_ID_ONE 0x0d
+#define BUTTON_ID_TWO 0x1f
+#define BUTTON_ID_S   0x07
+#define BUTTON_ID_M   0x0b
+
+/**
+SRSmith Pool Light Remote Control, Model #SRS-2C-TX
+
+The SR Smith remote control sends broadcasts of ~144 bits and it comes in shifted (similar to the Maverick XR30 BBQ Sensor)
+- Frequency: 915MHz
+
+Data Layout:
+    PPPP WWWW S UUUU C B T PP
+
+- P: 32 bit preamble (0xaaaaaaaa; 7 or 8 bits shifted left for analysis)
+- W: 32 bit sync word (0xd391d391)
+- S: 8 bit size (so far I've only seen 0x07)
+- U: 32 bit unknown (I always see 0x01fffff5 here)
+- C: 8 bit pin code is located in the bottom nibble of this byte, inverted and reversed.
+- B: 8 bit contains the ID of the button that was pushed on the remote
+- T: 8 bit CRC-8, poly 1, init 1 from the 16 bytes that we don't know (U) until the button that was pressed (B)
+- P: 16 bit CRC-16, poly 0x8005, init 0xFFFF, of the packet from the size (S) until the CRC-8 (T)
+
+Format String:
+    PRE:32h SYNC: 32h SIZE: hh UNSURE:32h | UNSURE: 4b | PIN ~^4b |  BTN: hh | CRC-8: hh | CRC-16: hhhh
+
+Capture raw but often misshifted :| data. if you want to work with this data, throw it into bitbench and shift it over until you see A's instead of 5's
+    -f 915000000 -X n=SRSmith,m=FSK_PCM,s=100,l=100,r=4096,match=d391d391
+*/
+
+static int srsmith_pool_srs_2c_tx_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+
+    if (bitbuffer->num_rows != 1)
+        return DECODE_ABORT_EARLY;
+
+    if (bitbuffer->bits_per_row[0] != 144)
+        return DECODE_ABORT_LENGTH;
+
+    // this is heavily derived from Maverick XR30 BBQ Sensor
+    uint8_t b[18];
+
+    // check for correct preamble/sync (0xaaaaaad391d391)
```

Comment:
It would improve robust reception to use `bitbuffer_search()` for just `aad391d391`